tests/poll.rs

```diff
@@ -201,6 +205,98 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+#[test]
+fn registry_operations_are_thread_safe() {
+    init();
+
+    let mut poll = Poll::new().unwrap();
+    let registry = Arc::new(poll.registry().try_clone().unwrap());
+    let mut events = Events::with_capacity(128);
+
+    let mut listener1 = TcpListener::bind(any_local_address()).unwrap();
+    let addr = listener1.local_addr().unwrap();
+    let mut listener2 = TcpListener::bind(any_local_address()).unwrap();
+    let barrier = Arc::new(Barrier::new(4));
+
+    let registry1 = Arc::clone(&registry);
+    let registry2 = Arc::clone(&registry);
+    let registry3 = Arc::clone(&registry);
+    let barrier1 = Arc::clone(&barrier);
+    let barrier2 = Arc::clone(&barrier);
+    let barrier3 = Arc::clone(&barrier);
+
+    let handle1 = thread::spawn(move || {
+        registry1
```

Comment:
Since these thread do mostly the same thing can you create another function for it to reduce the code duplication, some like:

```rust
/// Call all registration operations, ending with `source` being registered with `token` and `final_interests`.
fn registration_ops<S>(registry: &Registry, source: &mut io::Source, token: Token, init_interests: Interest, final_interests: Interest) -> io::Result<()> {
        registry.register(&mut source, token, init_interests).unwrap();
        registry.deregister(&mut source).unwrap();
        registry.register(&mut source, token, init_interests).unwrap();
        registry.reregister(&mut source, token, final_interests)
}
```

Reply:
great idea, will use it, thanks